collector/perf_schema_replication_applier_status_by_worker.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const perfReplicationApplierStatsByWorkerQuery = `
+	SELECT 
+	    CHANNEL_NAME,
+		WORKER_ID,
+		LAST_APPLIED_TRANSACTION_ORIGINAL_COMMIT_TIMESTAMP,
+		LAST_APPLIED_TRANSACTION_IMMEDIATE_COMMIT_TIMESTAMP,
+		LAST_APPLIED_TRANSACTION_START_APPLY_TIMESTAMP,
+		LAST_APPLIED_TRANSACTION_END_APPLY_TIMESTAMP,
+		APPLYING_TRANSACTION_ORIGINAL_COMMIT_TIMESTAMP,
+		APPLYING_TRANSACTION_IMMEDIATE_COMMIT_TIMESTAMP, 
+	  	APPLYING_TRANSACTION_START_APPLY_TIMESTAMP
+    FROM performance_schema.replication_applier_status_by_worker
+	`
+
+// Metric descriptors.
+var (
+	performanceSchemaReplicationApplierStatsByWorkerLastAppliedTransactionOriginalCommitSecondDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, performanceSchema, "last_applied_transaction_original_commit_second"),
+		"A timestamp shows when the last transaction applied by this worker was committed on the original master.",
+		[]string{"channel_name", "member_id"}, nil,
+	)
+
+	performanceSchemaReplicationApplierStatsByWorkerLastAppliedTransactionImmediateCommitSecondDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, performanceSchema, "last_applied_transaction_immediate_commit_second"),
```

Comment:
Minor typo, these should be plural, `_seconds`.

Reply:
This is why we usually call this `_time_seconds` or `_timestamp_seconds`.

Reply:
is `_timestamp_seconds` more fit for this case?

Reply:
Yes, I would say `_timestamp_seconds` is the best for this case.